smart_contracts/contract/src/contract_api/crypto.rs

```diff
@@ -0,0 +1,25 @@
+//! Functions with cryptographic utils.
+
+use casper_types::{api_error, HashAlgoType, BLAKE2B_DIGEST_LENGTH};
+
+use crate::{ext_ffi, unwrap_or_revert::UnwrapOrRevert};
+
+/// Computes digest hash, using provided algorithm type.
+pub fn generic_hash<T: AsRef<[u8]>>(input: T, algo: HashAlgoType) -> [u8; 32] {
+    let mut ret = [0; 32];
+
+    let algo_ptr = &algo as *const _ as *const u8;
+
+    let result = unsafe {
+        ext_ffi::casper_generic_hash(
+            input.as_ref().as_ptr(),
+            input.as_ref().len(),
+            algo_ptr,
+            1, // HashAlgoType is just one byte (u8).
```

Comment:
```suggestion
            algo as u8,
```

And no need for `algo_ptr` etc. `HashAlgoType` is safe to passed as an integer.

Reply:
Done in https://github.com/casper-network/casper-node/pull/4412/commits/2c70942af76e7e46b5c9d1fa74445d8d5131e555.